lib/local-links-manager/check_links/link_checker.rb

```diff
@@ -0,0 +1,52 @@
+module LocalLinksManager
+  module CheckLinks
+    class LinkChecker
+      TIMEOUT = 5
+      REDIRECT_LIMIT = 10
+
+      attr_reader :link_responses
+
+      def initialize
+        @link_responses = {}
+      end
+
+      def check_links(links)
+        links.each do |link|
+          link_responses[link] = [fetch_status(link), Time.zone.now]
+        end
+        link_responses
+      end
+
+    private
+
+      def fetch_status(link)
+        begin
+          response = connection.get(URI.parse(link)) do |request|
+            request.options[:timeout] = TIMEOUT
+            request.options[:open_timeout] = TIMEOUT
+          end
+          response.status.to_s
+        rescue Faraday::ConnectionFailed
+          "Connection failed"
+        rescue Faraday::TimeoutError
+          "Timeout Error"
+        rescue Faraday::SSLError
+          "SSL Error"
+        rescue FaradayMiddleware::RedirectLimitReached
+          "Too many redirects"
+        rescue URI::InvalidURIError
+          "Invalid URI"
+        rescue => e
+          e.class.to_s
+        end
+      end
+
+      def connection
+        @connection ||= Faraday.new(headers: { accept_encoding: 'none' }) do |faraday|
```

Comment:
So when we make a GET request to http://councillors.dudley.gov.uk, (without the encoding option) we get the following error:
```
Zlib::DataError: unknown compression method
```

Googling reveals this (https://github.com/lostisland/faraday/issues/120) and suggests to set that option as a workaround. It sounds like it's caused by combination of a Ruby bug and what the server is sending back? I guess there's a question here of whether we want to support this kind of thing or whether we want to flag this as an error?